server/async_tcp.go

```diff
@@ -250,6 +258,99 @@ func (s *AsyncServer) InitiateShutdown() {
 
 		delete(s.connectedClients, fd)
 	}
+}
+
+func (s *AsyncServer) executeCommandToBuffer(cmd *core.RedisCmd, buf *bytes.Buffer, c *core.Client) {
+	s.shardManager.Shards[0].ReqChan <- &core.StoreOp{
+		Cmd:      cmd,
+		WorkerID: "server",
+		ShardID:  0,
+		Client:   c,
+	}
+
+	response := <-s.ioChan
```

Comment:
Are we intentionally blocking and waiting for the response?

Reply:
This is similar to the iothread waiting for a response from the shard. The iothread has nothing else to do in the meantime, and this method's main purpose is to return whatever the result of the query is.